### PR TITLE
Handle multiple fastq objects in Sample.get_reads method

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

File: resdk/resources/sample.py [hunk 1]
```diff
@@ -15,9 +15,26 @@ class SampleUtilsMixin(object):
...
+        filtered, subsampled...), return the last one. If any other of
+        the ``fastq`` objects is required, one can provide additional
+        ``filter`` arguments and limits search to one result.
+        """
+        reads = self.data.filter(
+            type='data:reads:fastq',
+            ordering='-id',
+            **filters,
+        )
+        # TODO: In future, implement method ``last()`` on ResolweQuery
```
Comment (mstajdohar) [inline resdk/resources/sample.py]: Change 
```
# and use it instead or ordering by ``id``.
```
to
```
# and return the ResolweQuery object.
```

File: resdk/resources/sample.py [hunk 2]
```diff
@@ -15,9 +15,26 @@ class SampleUtilsMixin(object):
...
+
+        If there are multiple ``fastq`` objects in sample (trimmed,
+        filtered, subsampled...), return the last one. If any other of
+        the ``fastq`` objects is required, one can provide additional
+        ``filter`` arguments and limits search to one result.
+        """
+        reads = self.data.filter(
+            type='data:reads:fastq',
+            ordering='-id',
+            **filters,
```
Comment (mstajdohar) [inline resdk/resources/sample.py]: If user defines the `type` filter, we should override the type.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -21,6 +21,7 @@ Added
 
 Fixed
 -----
+* Handle multiple ``fastq`` objects in ``Sample.get_reads`` method
```
Comment (mstajdohar) [inline docs/CHANGELOG.rst]: Make more descriptive. Explain the latest read object will be returned.

File: resdk/resources/sample.py [hunk 4]
```diff
@@ -15,9 +15,26 @@ class SampleUtilsMixin(object):
...
+        )
+        # TODO: In future, implement method ``last()`` on ResolweQuery
+        # and use it instead or ordering by ``id``.
+
+        if not reads:
+            raise LookupError('Matching object does not exist.')
```
Comment (mstajdohar) [inline resdk/resources/sample.py]: I suggest something like `Reads not found on sample {}`

File: resdk/resources/sample.py [hunk 5]
```diff
@@ -15,9 +15,26 @@ class SampleUtilsMixin(object):
 
     """
 
-    def get_reads(self):
-        """Return ``fastq`` object on the sample."""
-        return self.data.get(type='data:reads:fastq')
+    def get_reads(self, **filters):
+        """Return ``fastq`` object in sample.
```
Comment (mstajdohar) [inline resdk/resources/sample.py]: Change to `Return the latest ``fastq`` object in sample.`

**Comment by mstajdohar** [comment]
Tests would be nice.

**Comment by JureZmrzlikar** [comment]
@mstajdohar  please have another review pass...

**Pull request merged**